policy-test/tests/e2e_authorization_policy.rs

```diff
@@ -48,6 +50,63 @@ async fn meshtls() {
     .await;
 }
 
+#[tokio::test(flavor = "current_thread")]
+async fn targets_route() {
+    with_temp_ns(|client, ns| async move {
+        // First create all of the policies we'll need so that the nginx pod
+        // starts up with the correct policy (to prevent races).
+        //
+        // The policy requires that all connections are authenticated with MeshTLS.
+        let (srv, all_mtls) = tokio::join!(
+            create(&client, nginx::server(&ns)),
+            create(&client, all_authenticated(&ns)),
+        );
+        // Create a route which matches the /allowed path.
+        let route = create(
+            &client,
+            http_route(&ns, &srv.name_unchecked(), NonZeroU16::new(80).unwrap()),
+        )
+        .await;
+        // Create a policy which allows all authenticated clients
+        create(
+            &client,
+            authz_policy(
+                &ns,
+                "nginx",
+                LocalTargetRef::from_resource(&route),
+                Some(NamespacedTargetRef::from_resource(&all_mtls)),
+            ),
+        )
+        .await;
+
+        // Create the nginx pod and wait for it to be ready.
+        tokio::join!(
+            create(&client, nginx::service(&ns)),
+            create_ready_pod(&client, nginx::pod(&ns))
+        );
+
+        let curl = curl::Runner::init(&client, &ns).await;
+        let (allowed, denied) = tokio::join!(
+            curl.run(
+                "curl-allowed",
+                "http://nginx/allowed",
+                LinkerdInject::Enabled
+            ),
+            curl.run("curl-denied", "http://nginx/denied", LinkerdInject::Enabled),
```

Comment:
This should probably target an allowed resource but not have sufficient authorization to show that authorization actually works:

```suggestion
            curl.run("curl-denied", "http://nginx/allowed", LinkerdInject::Disabled),
```